gitops/PrivateRepositories.md

```diff
@@ -0,0 +1,111 @@
+method: local
+source: component-files
+subcomponents:
+  - name: private-component
+    method: git
+    source: https://github.com/contoso/my-private-repository
+```
+
+At the same level, we introduce an `access.yaml` file with the following contents:
+
+```
+https://github.com/contoso/my-private-repository: ENV_VAR_CONTAINING_PAT
+```
+
+To test that this workflow works, you may:
+  1. Set the environment variable value `ENV_VAR_CONTAINING_PAT` to a Personal Access Token for the referenced repository (https://github.com/contoso/my-private-repository).
+  2. Invoke `fab install` from the same directory as the component.
+
+## SSH Keys
+
+While Personal Access Tokens offer a reliable way to read and clone Git repositories on behalf of a user, they can become unwieldy to use in practice. For an Azure DevOps scenario, imagine if a component's subcomponents were spread across multiple Organizations - a user supplying a Personal Access Token would have to provide one for every organization the component depends on. For example, given the following component.yaml:
+
+```
+name: episerver-cluster-hld
+type: component
+subcomponents:
+- name: private-component
+  type: component
+  source: https://dev.azure.com/contoso/cool-project/_git/private-component
+  method: git
+- name: awesome-component
+  type: component
+  source: https://dev.azure.com/fabrikam/awesome-project/_git/awesome-component
+  method: git
+```
+
+A corresponding access.yaml would like:
+
+```
+https://dev.azure.com/contoso/cool-project/_git/private-component: ENV_VAR_FOR_CONTOSO_PAT
+https://dev.azure.com/fabrikam/awesome-project/_git/awesome-component: ENV_VAR_FOR_FABRIKAM_PAT
+```
+
+And would require Personal Access Tokens for both the `contoso` organization and the `fabrikam` organization for a `fab install` to work as expected. Note that due to the recursive nature of components, a subcomponent can refer to other private git repositories as subcomponents as well. The subcomponent's root component.yaml can thus also provide an adjacent access.yaml, specifying the required Personal Access Tokens as environment variable names, and so on.
+
+Managing Personal Access Tokens can be utilized for simple components, but as you can imagine, the complexity increases dramatically as one adds more private components as subcomponents. As such, you may be interested in managing access to components via SSH Key Pairs, which many, if not all, Git providers provide as an authentication mechanism out of the box.
+
+To utilize SSH Key Pairs you must do the following:
+
+1. Utilizing a SSH toolkit ([we suggest OpenSSH](https://www.openssh.com/)), [generate an SSH Key Pair](https://www.ssh.com/ssh/keygen/).
+2. Upload the Public Key to Azure DevOps (the URL depends on your organization, but typically follows the pattern https://dev.azure.com/${organization}/_usersSettings/keys) or [GitHub](https://github.com/settings/keys).
+3. [Add the Private Key to the SSH Agent](https://www.ssh.com/ssh/add)
```

Comment:
unitalicize?

Reply:
I think this is a github rendering issue - the `_` in the url makes it look as if it's all intentionally italicized